test/WebJobs.Script.Tests/Configuration/HostHstsOptionsSetupTests.cs

```diff
@@ -0,0 +1,126 @@
+using Microsoft.Extensions.Options;
+using Microsoft.WebJobs.Script.Tests;
+using Moq;
+using Xunit;
+using static Microsoft.Azure.WebJobs.Script.EnvironmentSettingNames;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Configuration
+{
+    public class HostHstsOptionsSetupTests
+    {
+        private readonly TestEnvironment _environment = new TestEnvironment();
+        private readonly TestLoggerProvider _loggerProvider = new TestLoggerProvider();
+        private readonly string _hostJsonFile;
+        private readonly string _rootPath;
+        private readonly ScriptApplicationHostOptions _options;
+
+        public HostHstsOptionsSetupTests()
+        {
+            _rootPath = Path.Combine(Environment.CurrentDirectory, "ScriptHostTests");
+            Environment.SetEnvironmentVariable(AzureWebJobsScriptRoot, _rootPath);
+
+            if (!Directory.Exists(_rootPath))
+            {
+                Directory.CreateDirectory(_rootPath);
+            }
+
+            _options = new ScriptApplicationHostOptions
+            {
+                ScriptPath = _rootPath
+            };
+
+            _hostJsonFile = Path.Combine(_rootPath, "host.json");
+            if (File.Exists(_hostJsonFile))
+            {
+                File.Delete(_hostJsonFile);
+            }
+        }
+
+        [Theory]
+        [InlineData(@"{
+                    'version': '2.0',
+                    }")]
+        [InlineData(@"{
+                    'version': '2.0',
+                    'hsts' : { 
+                        'isEnabled' : true
+                        }
+                    }")]
+        public void MissingOrValidHstsConfig_DoesNotThrowException(string hostJsonContent)
+        {
+            File.WriteAllText(_hostJsonFile, hostJsonContent);
+            var configuration = BuildHostJsonConfiguration();
+
+            HostHstsOptionsSetup setup = new HostHstsOptionsSetup(configuration);
+            HostHstsOptions options = new HostHstsOptions();
+            var ex = Record.Exception(() => setup.Configure(options));
+            Assert.Null(ex);
+        }
+
+        [Theory]
+        [InlineData(@"{
+                    'version': '2.0',
+                    'hsts' : { 
+                        'maxAge' : '30'
+                        }
+                    }")]
+        public void HstsConfig_IsEnabledNotSet_Throws(string hostJsonContent)
+        {
+            File.WriteAllText(_hostJsonFile, hostJsonContent);
+            var configuration = BuildHostJsonConfiguration();
+
+            HostHstsOptionsSetup setup = new HostHstsOptionsSetup(configuration);
+            HostHstsOptions options = new HostHstsOptions();
+            var ex = Assert.Throws<ArgumentException>(() => setup.Configure(options));
+            Assert.StartsWith($"The value of IsEnabled property in hsts section of {ScriptConstants.HostMetadataFileName} file is missing. See https://aka.ms/functions-hostjson for more information", ex.Message);
```

Comment:
This should just be validating against the expected exception, instead of the message.